ESP32/main/src/gamepads/ps_3_4_controller.cpp

```diff
@@ -0,0 +1,540 @@
+      return Ps3.event.button_down.up;
+    #endif
+}
+
+int ps_3_4_controller::is_down_left()
+{
+    #ifdef IS_PS4
+      return PS4.event.button_down.left;
+    #endif
+    #ifdef IS_PS3
+      return Ps3.event.button_down.left;
+    #endif
+}
+
+int ps_3_4_controller::is_down_right()
+{
+    #ifdef IS_PS4
+      return PS4.event.button_down.right;
+    #endif
+    #ifdef IS_PS3
+      return Ps3.event.button_down.right;
+    #endif
+}
+
+int ps_3_4_controller::is_charging()
+{
+    #ifdef IS_PS4
+      return PS4.Charging();
+    #endif
+    #ifdef IS_PS3
+      return Ps3.Charging();
+    #endif
+}
+
+int ps_3_4_controller::is_connected()
+{
+    #ifdef IS_PS4
+      return PS4.isConnected();
+    #endif
+    #ifdef IS_PS3
+      return Ps3.isConnected();
+    #endif
+}
+
+int ps_3_4_controller::analog_ly()
+{
+    #ifdef IS_PS4
+      return PS4.data.analog.stick.ly;
+    #endif
+    #ifdef IS_PS3
+      return Ps3.data.analog.stick.ly;
+    #endif
+}
+
+int ps_3_4_controller::analog_lx()
+{
+    #ifdef IS_PS4
+      return PS4.data.analog.stick.lx;
+    #endif
+    #ifdef IS_PS3
+      return Ps3.data.analog.stick.lx;
+    #endif
+}
+
+int ps_3_4_controller::analog_ry()
+{
+    #ifdef IS_PS4
+      return PS4.data.analog.stick.ry;
+    #endif
+    #ifdef IS_PS3
+      return Ps3.data.analog.stick.ry;
+    #endif
+}
+
+int ps_3_4_controller::analog_rx()
+{
+    #ifdef IS_PS4
+      return PS4.data.analog.stick.rx;
+    #endif
+    #ifdef IS_PS3
+      return Ps3.data.analog.stick.rx;
+    #endif
+}
+
+int ps_3_4_controller::left()
+{
+    #ifdef IS_PS4
+      return PS4.data.button.left;
+    #endif
+    #ifdef IS_PS3
+      return Ps3.data.button.left;
+    #endif
+}
+
+int ps_3_4_controller::right()
+{
+    #ifdef IS_PS4
+      return PS4.data.button.right;
+    #endif
+    #ifdef IS_PS3
+      return Ps3.data.button.right;
+    #endif
+}
+
+/*int ps_3_4_controller::start_b()
+{
+    #ifdef IS_PS4
+      return PS4.data.button.start;
+    #endif
+    #ifdef IS_PS3
+      return Ps3.data.button.start;
+    #endif
+}*/
+
+int ps_3_4_controller::r2()
+{
+    #ifdef IS_PS4
+      return PS4.data.button.r2;
+    #endif
+    #ifdef IS_PS3
+      return Ps3.data.button.r2;
+    #endif
+}
+
+int ps_3_4_controller::l2()
+{
+    #ifdef IS_PS4
+      return PS4.data.button.l2;
+    #endif
+    #ifdef IS_PS3
+      return Ps3.data.button.l2;
+    #endif
+}
+
+int ps_3_4_controller::r1()
+{
+    #ifdef IS_PS4
+      return PS4.data.button.r1;
+    #endif
+    #ifdef IS_PS3
+      return Ps3.data.button.r1;
+    #endif
+}
+
+int ps_3_4_controller::l1()
+{
+    #ifdef IS_PS4
+      return PS4.data.button.l1;
+    #endif
+    #ifdef IS_PS3
+      return Ps3.data.button.l1;
+    #endif
+}
+
+
+void ps_3_4_controller::start()
+{
+    Serial.println("Connnecting to the PS4 controller, please please press the magic round button in the center..");
+    #ifdef IS_PS4
+      PS4.attach(ps_onAttach,this);
+      PS4.begin("1a:2b:3c:01:01:01 - UNICAST!");
```

Comment:
If we want to use the jjson parser to also include the mac-address, we should make this adaptble, right? Same with the PS3 controller. 

on error on my side anyway, should read:
`PS4.begin("1a:2b:3c:01:01:01")`

Reply:
yes. forgot to make it availible. maybe its better to use a extra endepoint. or drop that libs and use the bt hid interface